pkg/fanal/analyzer/language/java/pom/pom.go

```diff
@@ -21,10 +21,18 @@ func init() {
 const version = 1
 
 // pomAnalyzer analyzes pom.xml
-type pomAnalyzer struct{}
+type pomAnalyzer struct {
+	includeDevDeps bool
+}
 
-func (a pomAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
+func (a *pomAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
 	filePath := filepath.Join(input.Dir, input.FilePath)
+
+	// Skip integration test dir if `--include-dev-deps` flag is not present.
+	if isIntegrationTestDir(filePath) && !a.includeDevDeps {
+		return nil, nil
+	}
```

Comment:
I was thinking about moving this check to the `Required` function.
But it might not be entirely obvious